beater/telemetry.go

```diff
@@ -92,11 +86,26 @@ func recordConfigs(info beat.Info, apmCfg *config.Config, rootCfg *common.Config
 	configMonitors.ilmSetupRequirePolicy.Set(indexManagementCfg.ILM.Setup.RequirePolicy)
 	mode := indexManagementCfg.ILM.Mode
 	configMonitors.ilmEnabled.Set(mode == ilm.ModeAuto || mode == ilm.ModeEnabled)
+	return nil
+}
 
-	tailSamplingEnabled := apmCfg.Sampling.Tail != nil && apmCfg.Sampling.Tail.Enabled
-	configMonitors.tailSamplingEnabled.Set(tailSamplingEnabled)
+// recordAPMServerConfig records dynamic APM Server config properties for telemetry.
+// This should be called once each time a server is started.
```

Comment:
you mean the HTTP server, right? not the apm-server

Reply:
Yes. I literally meant the "server" type; I'll clarify the comment.